controllers/workloads/console/controller.go

```diff
@@ -635,6 +656,41 @@ func calculateStatus(csl *workloadsv1alpha1.Console, statusCtx consoleStatusCont
 	return newStatus
 }
 
+func (r *ConsoleReconciler) abort(ctx context.Context, logger logr.Logger, csl *workloadsv1alpha1.Console, job *batchv1.Job, podList *corev1.PodList) error {
+	// Delete job
+	if err := r.Client.Delete(ctx, job); err != nil {
+		return errors.Wrap(err, "failed to delete job")
+	}
+
+	// Delete pods
+	var podDeleteError error
+	for _, pod := range podList.Items {
+		if err := r.Client.Delete(ctx, &pod); err != nil {
+			podDeleteError = err
+		}
+	}
+	if podDeleteError != nil {
+		return errors.Wrap(podDeleteError, "failed to delete pod(s)")
```

Comment:
I'm not sure we need this, as the Job will own the pods. Deleting the job should remove all of the pods for us.

Reply:
IIRC there's no guarantee that the job controller will get the deletion event prior to being notified of it's pods being removed, although  this is unlikely.

Reply:
Yes indeed, in theory we shouldn't have to delete pods. They are owned by the console job. A delete operation should cascade. However, when I tested manual deletion, for some reason the newly launched second pod lingers on indefinitely after the job is gone.

Reply:
As agreed, I’ve added a comment in the source code about this.